src/test/java/by/radioegor146/source/StringPoolTest.java

```diff
@@ -0,0 +1,51 @@
+package by.radioegor146.source;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringPoolTest {
+
+    @Test
+    public void testBuild(){
+        StringPool stringPool = new StringPool();
+
+        stringPool.get("test");
+
+        Assert.assertEquals(
+            "#include \"string_pool.hpp\"\n" +
+            "\n" +
+            "namespace native_jvm::string_pool {\n" +
+            "    static char pool[5LL] = { 116, 101, 115, 116, 0 };\n" +
+            "\n" +
+            "    char *get_pool() {\n" +
+            "        return pool;\n" +
+            "    }\n" +
+            "}", stringPool.build());
+
+        stringPool.get("other");
+
+        Assert.assertEquals(
+            "#include \"string_pool.hpp\"\n" +
+            "\n" +
+            "namespace native_jvm::string_pool {\n" +
+            "    static char pool[11LL] = { 116, 101, 115, 116, 0, 111, 116, 104, 101, 114, 0 };\n" +
+            "\n" +
+            "    char *get_pool() {\n" +
+            "        return pool;\n" +
+            "    }\n" +
+            "}", stringPool.build());
+    }
+
+    @Test
+    public void testGet(){
+        StringPool stringPool = new StringPool();
+
+        Assert.assertEquals("((char *)(string_pool + 0LL))", stringPool.get("test"));
```

Comment:
These two asserts look identical (to my eyes) Is that intentional?

Reply:
Yeah, the behaviour is different if you call it twice with the same string (even though the return is the same).

Reply:
Ah ok. Makes sense.